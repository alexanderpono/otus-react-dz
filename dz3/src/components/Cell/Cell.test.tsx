import React from 'react';
import { shallow } from 'enzyme';
import { Cell, SHOW, HIDE } from './';
import renderer from 'react-test-renderer';

describe('Cell', () => {
    it('It renders using shallow', () => {
        const wrapper = shallow(<Cell num={3}></Cell>);
        expect(wrapper.html()).toEqual('<article class="game-cell"><span>3</span></article>');
    });

    it('It renders using renderer and is compared to snaphot', () => {
        const snapshot = renderer.create(<Cell num={2}></Cell>).toJSON();
        expect(snapshot).toMatchSnapshot();
    });

    it('OnClick callback is called when cell is clicked', () => {
        const mockCallBack = jest.fn();
        const cell = shallow(<Cell num={2} onClick={mockCallBack}></Cell>);
        cell.find('article').simulate('click');
        expect(mockCallBack.mock.calls.length).toEqual(1);
    });

    it('Says number of cell and says show state when cell is clicked', () => {
        let receivedCellNumber = 0;
        let receivedCellState = '';
        const testCallBack = function (cellNumber: number, newState: string) {
            receivedCellNumber = cellNumber;
            receivedCellState = newState;
        };
        const cell = shallow(<Cell num={2} onClick={testCallBack}></Cell>);
        cell.find('article').simulate('click');
        expect(receivedCellNumber).toEqual(2);
        expect(receivedCellState).toEqual(SHOW);

        cell.find('article').simulate('click');
        expect(receivedCellNumber).toEqual(2);
        expect(receivedCellState).toEqual(HIDE);
    });
});
