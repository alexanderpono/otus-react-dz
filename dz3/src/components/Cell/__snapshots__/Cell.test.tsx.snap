// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell It renders using renderer and is compared to snaphot 1`] = `
<article
  className="game-cell"
  onClick={[Function]}
>
  <span>
    2
  </span>
</article>
`;
